feather/base/src/anvil/entity.rs

```diff
@@ -236,11 +238,19 @@ pub struct ItemNbt {
 
 impl ItemNbt {
     /// Create an `ItemStack` of the specified item and amount, setting any nbt present.
+    /// Panics if count is zero.
     pub fn item_stack(nbt: &Option<Self>, item: Item, count: u8) -> ItemStack {
-        ItemStack {
-            count: count as u32,
-            item,
-            damage: nbt.as_ref().map(|n| n.damage).flatten().map(|x| x as u32),
+        match nbt {
+            Some(ItemNbt {
+                damage: Some(damage),
+            }) => ItemStackBuilder::with_item(item)
+                .count(count as u32)
+                .damage((*damage).try_into().unwrap())
```

Comment:
Is unwrapping here okay? Should we be returning a `Result` or can this never panic?

Reply:
Nice catch, this is an issue.  I am going to change this.

Reply:
I made damage on ItemStack be a i32, reflecting the fact that vanilla minecraft allows negative damage amounts, making the anvil damage and itemstack damage 
congruent.